ddlitlab2024/dataset/models.py

```diff
@@ -171,32 +173,32 @@ class JointState(Base):
-    l_hip_roll: Mapped[float] = mapped_column(Float, nullable=False, name="LHipRoll")
-    r_hip_pitch: Mapped[float] = mapped_column(Float, nullable=False, name="RHipPitch")
-    l_hip_pitch: Mapped[float] = mapped_column(Float, nullable=False, name="LHipPitch")
-    r_knee: Mapped[float] = mapped_column(Float, nullable=False, name="RKnee")
-    l_knee: Mapped[float] = mapped_column(Float, nullable=False, name="LKnee")
-    r_ankle_pitch: Mapped[float] = mapped_column(Float, nullable=False, name="RAnklePitch")
-    l_ankle_pitch: Mapped[float] = mapped_column(Float, nullable=False, name="LAnklePitch")
-    r_ankle_roll: Mapped[float] = mapped_column(Float, nullable=False, name="RAnkleRoll")
-    l_ankle_roll: Mapped[float] = mapped_column(Float, nullable=False, name="LAnkleRoll")
-    head_pan: Mapped[float] = mapped_column(Float, nullable=False, name="HeadPan")
-    head_tilt: Mapped[float] = mapped_column(Float, nullable=False, name="HeadTilt")
+    r_shoulder_pitch: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="RShoulderPitch")
+    l_shoulder_pitch: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="LShoulderPitch")
+    r_shoulder_roll: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="RShoulderRoll")
+    l_shoulder_roll: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="LShoulderRoll")
+    r_elbow: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="RElbow")
+    l_elbow: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="LElbow")
+    r_hip_yaw: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="RHipYaw")
+    l_hip_yaw: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="LHipYaw")
+    r_hip_roll: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="RHipRoll")
+    l_hip_roll: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="LHipRoll")
+    r_hip_pitch: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="RHipPitch")
+    l_hip_pitch: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="LHipPitch")
+    r_knee: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="RKnee")
+    l_knee: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="LKnee")
+    r_ankle_pitch: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="RAnklePitch")
+    l_ankle_pitch: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="LAnklePitch")
+    r_ankle_roll: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="RAnkleRoll")
+    l_ankle_roll: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="LAnkleRoll")
+    head_pan: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="HeadPan")
+    head_tilt: Mapped[Optional[float]] = mapped_column(Float, nullable=True, name="HeadTilt")
```

Comment:
I am not so sure if we want to allow this. Why do you want to make it optional?

Reply:
Because `JointCommands` only ever contain the motor commands for the motors, where we actually want to change the position. Otherwise, the data does not even exist in the message.
E.g. if the head is not moved, the `HeadTilt` is empty.
This is different from `JointPositions` where we always have all positions.

But with resampling etc. it makes sense to use previous values or a default of 0.
I would do this in a seperate PR though.